LEAF_Request_Portal/templates/reports/LEAF_inbox_combined.tpl

```diff
@@ -1,128 +1,134 @@
+            }
         }
-    }
-    return icon;
-}
 
-function buildDepInbox(res, depID, site) {
-    var hash = Sha1.hash(site.url);
-    var dependencyName = res[depID].dependencyDesc;
-    if(String(depID).substr(0,2) == '-1') {
-        dependencyName = res[depID].approverName != null ? res[depID].approverName : 'Person designated by requestor';
+        $('#loading').slideUp();
+        $('#inboxContainer').fadeIn();
     }
 
-    var icon = getIcon(site.icon, site.name);
-    if(document.getElementById('siteContainer'+ hash) == null) {
-        $('#indexSites').append('<li style="font-size: 120%; line-height: 150%"><a href="#'+ hash +'">' + site.name + '</a></li>');
-        $('#inbox').append('<div id="siteContainer'+ hash +'" style="border-left: 4px solid '+ site.backgroundColor +'; margin: 0px auto 8px">'
-                            + '<a name="'+ hash +'" />'
-                            + '<div style="font-weight: bold; font-size: 200%; line-height: 240%; background-color: '+ site.backgroundColor +'; color: '+ site.fontColor +'; ">' + icon + ' ' + site.name + '</div>'
-                            + '</div>');
+    function getIcon(icon, name) {
+        if(icon != '') {
+            if(icon.indexOf('/') != -1) {
+                icon = '<img src="'+ icon +'" alt="icon for '+ name +'" style="vertical-align: middle" />';
+            }
+            else {
+                icon = '<img src="../libs/dynicons/?img='+ icon +'&w=76" alt="icon for '+ name +'" style="vertical-align: middle" />';
+            }
+        }
+        return icon;
     }
 
-    $('#siteContainer'+ hash).append('<div id="depContainer'+ hash +'_' + depID + '" style="border: 1px solid black; background-color: '+ res[depID].dependencyBgColor +'; cursor: pointer; margin: 4px">'
-                       + '<div id="depLabel'+ hash +'_'+ depID + '" class="depInbox" style="padding: 8px"><span style="float: right; text-decoration: underline; font-weight: bold">View '+ res[depID].count +' requests</span>'
-                       + '<span style="font-size: 120%; font-weight: bold">'+ dependencyName +'</span>'
-                       + '</div>'
-                       + '<div id="depList'+ hash +'_' + depID + '" style="width: 90%; margin: auto; display: none"></div></div><br />');
-    $('#depLabel'+ hash +'_'+ depID).on('click', function() {
-        if($('#depList'+ hash +'_'+ depID).css('display') == 'none') {
-            $('#depList'+ hash +'_'+ depID).css('display', 'inline');
+    function buildDepInbox(res, depID, site) {
+        var hash = Sha1.hash(site.url);
+        var dependencyName = res[depID].dependencyDesc;
+        if(String(depID).substr(0,2) == '-1') {
+            dependencyName = res[depID].approverName != null ? res[depID].approverName : 'Person designated by requestor';
         }
-        else {
-            $('#depList'+ hash +'_'+ depID).css('display', 'none');
+
+        var icon = getIcon(site.icon, site.name);
```

Comment:
This is setting a global variable "icon".  This name is used in many places to reference different objects.  If it's not your intention to set a global, use "let" instead of "var".